src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs

```diff
@@ -115,22 +115,27 @@ public class TextLayout
         /// Draws the text layout.
         /// </summary>
         /// <param name="context">The drawing context.</param>
-        /// <param name="origin">The origin.</param>
-        public void Draw(DrawingContext context, Point origin)
+        public void Draw(DrawingContext context)
         {
             if (!TextLines.Any())
             {
                 return;
             }
 
-            var currentY = origin.Y;
+            var currentY = 0.0;
 
             foreach (var textLine in TextLines)
             {
-                var offsetX = TextLine.GetParagraphOffsetX(textLine.LineMetrics.Size.Width, Size.Width,
-                    _paragraphProperties.TextAlignment);
+                using (context.PushPostTransform(Matrix.CreateTranslation(0, currentY)))
+                {
+                    var offsetX = TextLine.GetParagraphOffsetX(textLine.LineMetrics.Size.Width, Size.Width,
```

Comment:
Can we move it before first push post transform and push combined translation?

Reply:
Y good catch. That will be cleaner.